bg_atlasgen/validate_atlases.py

```diff
@@ -0,0 +1,195 @@
+    mesh_points = root_mesh.points
+    z_coords, y_coords, x_coords = (
+        mesh_points[:, 0],
+        mesh_points[:, 1],
+        mesh_points[:, 2],
+    )
+
+    # minimum and maximum coordinates of the root mesh
+    z_min_mesh, z_max_mesh = np.min(z_coords), np.max(z_coords)
+    y_min_mesh, y_max_mesh = np.min(y_coords), np.max(y_coords)
+    x_min_mesh, x_max_mesh = np.min(x_coords), np.max(x_coords)
+
+    # checking if root mesh and image are on the same scale
+    _assert_close(z_min_mesh, z_min_scaled, resolution[0])
+    _assert_close(z_max_mesh, z_max_scaled, resolution[0])
+    _assert_close(y_min_mesh, y_min_scaled, resolution[1])
+    _assert_close(y_max_mesh, y_max_scaled, resolution[1])
+    _assert_close(x_min_mesh, x_min_scaled, resolution[2])
+    _assert_close(x_max_mesh, x_max_scaled, resolution[2])
+
+    return True
+
+
+def open_for_visual_check():
+    pass
+
+
+def validate_checksum():
+    pass
+
+
+def check_additional_references():
+    # check additional references are different, but have same dimensions
+    pass
+
+
+def validate_atlas(atlas_name, version, all_validation_functions):
+    """Validates the latest version of a given atlas"""
+
+    print(atlas_name, version)
+    BrainGlobeAtlas(atlas_name)
+    updated = get_atlases_lastversions()[atlas_name]["updated"]
+    if not updated:
+        update_atlas(atlas_name)
+    Path(get_brainglobe_dir()) / f"{atlas_name}_v{version}"
+
+    validation_function_parameters = [
+        # validate_atlas_files(atlas_path: Path)
+        (Path(get_brainglobe_dir() / f"{atlas_name}_v{version}"),),
+        # validate_mesh_matches_image_extents(atlas: BrainGlobeAtlas)
+        (BrainGlobeAtlas(atlas_name),),
+        # open_for_visual_check()
+        (),
+        # validate_checksum()
+        (),
+        # check_additional_references()
+        (),
+        # validate_atlas(atlas_name, version)
+        (atlas_name, version),
+    ]
+
+    # list to store the errors of the failed validations
+    failed_validations = []
+    successful_validations = []
+
+    for i, validation_function in enumerate(all_validation_functions):
+        try:
+            validation_function(*validation_function_parameters[i])
+            successful_validations.append((atlas_name, validation_function))
+        except AssertionError as error:
+            failed_validations.append((atlas_name, validation_function, error))
+
+    return successful_validations, failed_validations
+
+
+def validate_mesh_structure_pairs(atlas_path: Path):
+    json_path = Path(atlas_path / "structures.json")
+    obj_path = Path(atlas_path / "meshes")
+
+    with open(json_path, "r") as file:
+        json_file = json.load(file)
+
+    obj_file_list = [
+        file for file in os.listdir(obj_path) if file.endswith(".obj")
+    ]
+    target_key = "id"
+    id_numbers = [item[target_key] for item in json_file if target_key in item]
+
```

Comment:
why do we need the `if` part here? Should not all items in the json_file have an `id` key?
I wonder whether we should check the id via the bg_atlasapi via something like
```python
atlas = BrainGlobeAtlas(atlas_name)
bg_atlas_api_ids = list(atlas.structures.keys())

# a fancy, one-line way to get the ids from the meshes - see https://docs.python.org/3/library/pathlib.html?highlight=stem#pathlib.PurePath.stem
ids_from_mesh_files = [
        int(Path(file).stem) for file in os.listdir(obj_path) if file.endswith(".obj")
    ] 

in_mesh_not_bg = []
for id in ids_from_mesh_files:
    if id not in bg_atlas_api_ids:
        in_mesh_not_bg.append(id)

in_bg_not_mesh = []
for id in bg_atlas_api_ids:
    if id not in ids_from_mesh_files:
        in_bg_not_mesh.append(id)

if len(in_mesh_not_bg) or len(in_bg_not_mesh):
    raise AssertionError(f"Structures with ID {in_bg_not_mesh} are in the atlas, but don't have a corresponding mesh file; Structures with IDs {in_mesh_not_bg} have a mesh file, but are not accessible through the atlas.")
```

This is designed to throw at most one assertion error, so successful and failed validation results get logged in the same way as other validation functions (see next comment).